service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.service.impl;
+
+import java.util.*;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbioportal.model.DataAccessToken;
+import org.cbioportal.persistence.DataAccessTokenRepository;
+import org.cbioportal.service.DataAccessTokenService;
+import org.cbioportal.service.exception.InvalidDataAccessTokenException;
+import org.cbioportal.service.util.JwtUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Service
+@Component
+public class UuidDataAccessTokenServiceImpl implements DataAccessTokenService {
+
+    @Autowired
+    private DataAccessTokenRepository dataAccessTokenRepository;
+
+    @Value("${jwt.ttl_seconds}")
+    private int datTtlSeconds;
+
+    @Value("${max.number.dat}")
+    private int maxNumberOfAccessTokens;
+
+    private static final Log log = LogFactory.getLog(UuidDataAccessTokenServiceImpl.class);
+
+    // create a data access token (randomly generated UUID) and insert corresponding record into table with parts:
+    // username
+    // uuid
+    // expiration date (current time + 1 month)
+    @Override
+    public DataAccessToken createDataAccessToken(String username, boolean allowRevocationOfOtherTokens) {
+        if (username == null || username.trim().length() == 0) {
+            throw new IllegalArgumentException("subject cannot be empty");
+        }
+        String uuid = UUID.randomUUID().toString();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.SECOND, datTtlSeconds);
+        Date expirationDate = calendar.getTime();
+       
+        DataAccessToken dataAccessToken = new DataAccessToken(); 
+        dataAccessToken.setToken(uuid);
+        dataAccessToken.setUsername(username);
+        dataAccessToken.setExpiration(expirationDate);
+        if (getNumberOfTokensForUsername(username) == maxNumberOfAccessTokens) {
+            revokeOldestDataAccessTokenForUsername(username);
```

Comment:
Here is where the argument to the function should be used. If allowRevocationOfOtherTokens is false, we should not revoke this token, and should fail with a thrown exception (maybe we need a new exception class like "DataAccessTokenUnableToCreateException")
Also, maybe this revoke logic should happen before we make the effort to generate a new UUID